common/protob/messages-solana.proto

```diff
@@ -0,0 +1,61 @@
+syntax = "proto2";
+package hw.trezor.messages.solana;
+
+import "messages-common.proto";
+
+/**
+ * Request: Ask device for public key corresponding to address_n path
+ * @start
+ * @next SolanaPublicKey
+ * @next Failure
+ */
+message SolanaGetPublicKey {
+    repeated uint32 address_n = 1;  // BIP-32 path to derive the key from master node
+    optional bool show_display = 2; // optionally show on display before sending the result
+}
+
+/**
+ * Response: Contains public key derived from device private seed
+ * @end
+ */
+message SolanaPublicKey {
+    required bytes public_key = 1;
+}
+
+/**
+ * Request: Ask device for Solana address
+ * @start
+ * @next SolanaAddress
+ * @next Failure
+ */
+message SolanaGetAddress {
+    repeated uint32 address_n = 1;  // BIP-32 path to derive the key from master node
+    optional bool show_display = 2; // optionally show on display before sending the result
```

Comment:
Added to `GetAddress` here -> https://github.com/trezor/trezor-firmware/pull/3359/commits/dd6058f533ad1ac07300169a3f470be6973b125a I'd rather not add it to sign tx since the UI is rather generic - but if you insist, we could probably handle it for all addresses.